src/election.ts

```diff
@@ -0,0 +1,167 @@
+import BigNumber from 'bignumber.js';
+import { Lease } from './lease';
+import { Namespace } from './namespace';
+
+/**
+ * Implmentation of etcd election.
+ * @see https://github.com/coreos/etcd/blob/master/clientv3/concurrency/election.go
+ *
+ * @example
+ * const client = new Etcd3()
+ * const election = new Election(client, 'singleton_service')
+ * const id = BigNumber.random().toString()
+ *
+ * // waiting for election ready
+ * await election.ready()
+ *
+ * // process will hang here until elected
+ * await election.campaign(id)
+ */
+export class Election {
+  public static readonly prefix = 'election';
+  public static readonly notLeaderError = new Error('election: not leader');
+  public static readonly noLeaderError = new Error('election: no leader');
+  public static readonly notReadyError = new Error('election: no ready');
```

Comment:
Rather than using sentinel errors like the Go client does, it'd be nice to make each of these subclasses of Error in `errors.ts` and instantiate them anew when thrown. Amongst other benefits, this make the stacktrace be set correctly; as-is the stacktrace will be set when the consumer imports the class, rather than when it's thrown!